sycl/include/sycl/khr/group_interface.hpp

```diff
@@ -0,0 +1,266 @@
+class sub_group;
+template <typename ParentGroup> class member_item;
+
+} // namespace khr
+
+namespace detail {
+#if defined(__cpp_lib_mdspan)
+template <typename IndexType, int Dimensions> struct single_extents;
+
+template <typename IndexType> struct single_extents<IndexType, 1> {
+  using type = std::extents<IndexType, 1>;
+};
+
+template <typename IndexType> struct single_extents<IndexType, 2> {
+  using type = std::extents<IndexType, 1, 1>;
+};
+
+template <typename IndexType> struct single_extents<IndexType, 3> {
+  using type = std::extents<IndexType, 1, 1, 1>;
+};
+#endif
+
+template <typename T> struct is_khr_group : public std::false_type {};
+
+template <int Dimensions>
+struct is_khr_group<khr::work_group<Dimensions>> : public std::true_type {};
+
+template <> struct is_khr_group<khr::sub_group> : public std::true_type {};
+
+} // namespace detail
+
+namespace khr {
+
+// Forward declaration for friend function.
+template <typename ParentGroup>
+std::enable_if_t<detail::is_khr_group<ParentGroup>::value,
+                 member_item<ParentGroup>>
+get_member_item(ParentGroup g);
+
+template <int Dimensions = 1> class work_group {
+public:
+  using id_type = sycl::id<Dimensions>;
+  using linear_id_type = size_t;
+  using range_type = sycl::range<Dimensions>;
+#if defined(__cpp_lib_mdspan)
+  using extents_type = std::dextents<size_t, Dimensions>;
+#endif
+  using size_type = size_t;
+  static constexpr int dimensions = Dimensions;
+  static constexpr memory_scope fence_scope = memory_scope::work_group;
+
+  work_group(group<Dimensions>) noexcept {}
+
+  operator group<Dimensions>() const noexcept { return legacy(); }
+
+  id_type id() const noexcept { return legacy().get_group_id(); }
+
+  linear_id_type linear_id() const noexcept {
+    return legacy().get_group_linear_id();
+  }
+
+  range_type range() const noexcept { return legacy().get_group_range(); }
+
+#if defined(__cpp_lib_mdspan)
+  constexpr extents_type extents() const noexcept {
+    auto LocalRange = legacy().get_local_range();
+    if constexpr (dimensions == 1) {
+      return extents_type(LocalRange[0]);
+    } else if constexpr (dimensions == 2) {
+      return extents_type(LocalRange[0], LocalRange[1]);
+    } else if constexpr (dimensions == 3) {
+      return extents_type(LocalRange[0], LocalRange[1], LocalRange[2]);
+    }
+  }
+
+  constexpr typename extents_type::index_type
+  extent(typename extents_type::rank_type r) const noexcept {
+    return extents().extent(r);
+  }
+
+  static constexpr typename extents_type::rank_type rank() noexcept {
+    return extents_type::rank();
+  }
+
+  static constexpr typename extents_type::rank_type rank_dynamic() noexcept {
+    return extents_type::rank_dynamic();
+  }
+
+  static constexpr size_t
+  static_extent(typename extents_type::rank_type r) noexcept {
+    return extents_type::static_extent(r);
+  }
+#endif
+
+  size_type size() const noexcept { return legacy().get_local_range().size(); }
```

Comment:
in doc - expected to be constexpr. could you please clarify if doc or impl should be updated?

Reply:
That is the intent.  It looks like I missed some: there are a few places where I removed the `constexpr` from the header but not from the function definition.  I'll fix that now, sorry for the confusion.

Reply:
Fixed in KhronosGroup/SYCL-Docs@1511696b3e6dcbc0bb96a4eba2a10fcdff447f27.